R/plotFeatureLoadings.R

```diff
@@ -0,0 +1,404 @@
+              rownames(df) <- phylo$tip.label
+              
+              
+              df2 <- data.frame(abs(loadings_matrix))
+              rownames(df2) <- phylo$tip.label
+              
+              
+              p <- gheatmap(
+                p = circ,
+                data = df,
+                offset = -.1,
+                width = .1,
+                colnames_angle = 95,
+                colnames_offset_y = .5,
+                font.size = 4,
+                color = "black") +
+                ggplot2::scale_fill_manual(
+                  values = color,
+                  name = "Class"
+                )
+              
+              for(i in 1:2){
+                if(i == 1){
+                  p <- p +
+                    ggnewscale::new_scale_fill()
+                }
+                df3 <- dplyr::select(
+                  df2, (all_of(i))
+                )
+                
+                p <- gheatmap(
+                  p,
+                  df3,
+                  offset = i*.065,
+                  width = .1,
+                  colnames_angle = 90,
+                  font.size = 4,
+                  high = "darkslateblue",
+                  low = "gray98",
+                  color = "black",
+                  legend_title = expression(beta[k]))
+              }
+              
+              p
+              
+            } else {
+              .plot_pca_feature_loadings(df, plot)
+            }
+          }
+)
+
+setMethod("plotFeatureLoadings", signature = c(object = "matrix"),
+          function(object,
+                   method = "PCA", plot = "Heatmap",
+                   ...){
+            
+            .check_parameters(object,
+                              method = method, plot = plot, ...)
+                        
+            if (method == "PCA") {
+              df <- data.frame(object)
+              df[["Feature"]] <- rownames(object)
+              df[["Feature"]] <- factor(df[["Feature"]], levels = df[["Feature"]])
+              df <- df[ order(abs(df[["PC1"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC1"]]), ]
+              df <- df[ order(abs(df[["PC2"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC2"]]), ]
+              .plot_pca_feature_loadings(df, plot)
+            } else {
+              stop("These methods are yet to be implemented", call. = FALSE)
+            }
+            
+
+          }
+)
+
+.check_parameters <- function(object, method, plot, ...) {
+  
+  if ( !(method %in% c("PCA", "LDA", "NMF")) ) {
+    stop("'method' must be one of c('PCA', 'LDA', 'NMF').", call. = FALSE)
+  }
+  
+  if ( !(plot %in% c("Screeplot", "Barplot", "Tree", "Heatmap")) ) {
+    stop("'method' must be one of c('Screeplot', 'Barplot', 'Tree', 'Heatmap').", call. = FALSE)
+  }
+  
+  if (is.matrix(object) && plot == "Tree") {
+    stop("Tree cannot be plotted for a non-TSE object.", call. = FALSE)
```

Comment:
Write positively: "TreeSummarizedExperiment object is required for the tree plotting".